### PR TITLE
Add Hidden Maze Infinite

### PR DESCRIPTION
**Pull request by Dark-Sniper101**
Author: 
Rylan Berry

About:
This game Hidden Maze Infinite (or Hidden Maze 2 if you want) is the newest game in my planned trilogy of hidden maze games, if you recall my hidden_maze that I shipped earlier in the month that game had a set amount of levels (5) which you had to get through to beat the game, now you have an infinite amount of levels that you can play through, complete one you move onto the next, each maze will be the same size (but later I may update that).

How to Play:
Same as the 1st game, WASD to move, and you have to get to the goal to progress to the next level.

Code
- [ X] The game was made using the [Sprig editor](https://sprig.hackclub.com/editor).
- [X ] The game is placed in the in the [`/games` directory](https://github.com/hackclub/sprig/tree/main/games).
- [X ] The code is significantly different from all other games in the [Sprig gallery](https://sprig.hackclub.com/gallery) (except for games labeled "demo").
- [X ] The game runs without errors. 
- [ X] The name of the file/game contains only alphanumeric characters, `-`s, or `_`s.
- [X ] The game name is not the same as the others from [gallery](https://sprig.hackclub.com/gallery)

### CODE REVIEW

File: games/hidden_maze_infinite.js [hunk 1]
```diff
@@ -0,0 +1,297 @@
+/*
+Welcome to Hidden Maze Infinite (or Hidden Maze 2 if you will), the 2nd game in the planned trilogoy of my Hidden Maze games here one sprig, 
+I would recomend playing the 1st game, its a short introductory for what is to come.
+Controls are the same WASD, don't bother with worrying about getting to the end in this one, there is none, 
+I have carefully constructed a maze generation that will allow you to play for as long as you like.
+Remeber, have fun!
```
Comment (LucasHT22) [inline games/hidden_maze_infinite.js:6]: missing an M here:)

File: games/hidden_maze.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
+/*
```
Comment (LucasHT22) [inline games/hidden_maze.js:1]: this version is already in the gallery, right? Then delete it so the file doesn't get duplicated

**Review by LucasHT22 — CHANGES_REQUESTED** [review]
Awessome! Infinite it's cool:)

File: games/hidden_maze_infinite.js (hunk 1 above)
Comment (Dark-Sniper101) [reply games/hidden_maze_infinite.js]: fixed(should be)

File: games/hidden_maze.js (hunk 2 above)
Comment (Dark-Sniper101) [reply games/hidden_maze.js]: Should be deleted, this is only my second time making a pull request, and only my second time making a 2nd pull request from the same fork.

Comment (LucasHT22) [reply games/hidden_maze.js]: no problem!:)

**Review by LucasHT22 — APPROVED** [review]
Congrats for your 2nd game!

**Pull request merged**